server/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -385,14 +387,6 @@ public void onNewClusterStateFailed(Exception e) {
                 return;
```

Comment:
This, and the containing `synchronised` block, don't look right. They throw `FailedToCommitClusterStateException` rather than passing it to the `publishListener`, and previously they returned early without blocking on the publication but the equivalent flow now would be to call `publishListener.onResponse(null)` early somehow.

Reply:
fixed in 506608c